spec/surfnperf_spec.js

```diff
@@ -1096,5 +1096,108 @@ define('spec/surfnperf_spec', [
+      it('returns a reference to window.performance.timing.msFirstPaint if the browser supports it, otherwise undefined', function() {
+        if(window.performance && window.performance.timing && window.performance.timing.msFirstPaint) {
+          expect(SurfNPerf.msFirstPaint()).toEqual(window.performance.timing.msFirstPaint);
+        } else {
+          expect(SurfNPerf.msFirstPaint()).toEqual(undefined);
+        }
+      });
+    });
+
+    describe('#getFirstPaint', function() {
+      describe('when the client is using Chrome', function() {
+        it('returns the time to first paint relative to navigation start time', function() {
+          spyOn(SurfNPerf, 'chromeLoadTimes').andReturn({
+            firstPaintTime: 1446480965.123456
+          });
+          spyOn(SurfNPerf, 'msFirstPaint').andReturn(undefined);
+          spyOn(SurfNPerf, 'performanceTiming').andReturn({
+            navigationStart: 123789.734
+          });
+
+          expect(SurfNPerf.getFirstPaint()).toEqual(1446480965.123456 * 1000 - SurfNPerf.performanceTiming().navigationStart);
+        });
+      });
+
+      describe('when the client is using IE/Edge', function() {
+        it('returns the time to first paint relative to navigation start time', function() {
+          spyOn(SurfNPerf, 'msFirstPaint').andReturn(1776480965.123457);
+          spyOn(SurfNPerf, 'chromeLoadTimes').andReturn(undefined);
+          spyOn(SurfNPerf, 'performanceTiming').andReturn({
+            navigationStart: 123789.734
+          });
+
+          expect(SurfNPerf.getFirstPaint()).toEqual(SurfNPerf.msFirstPaint() - SurfNPerf.performanceTiming().navigationStart);
+        });
+      });
+
+      describe('when the client is using browser without Time to First Paint built in support', function() {
+        it('returns null since the browser does not have built in support', function() {
+          spyOn(SurfNPerf, 'chromeLoadTimes').andReturn(undefined);
+          spyOn(SurfNPerf, 'msFirstPaint').andReturn(undefined);
+          spyOn(SurfNPerf, 'performanceTiming').andReturn({
+            navigationStart: 123789.734
+          });
+
+          expect(SurfNPerf.getFirstPaint()).toEqual(null);
+        });
+      });
+    });
+
+    describe("#getFirstPaintFrame", function() {
+      describe('when the client does not support window.requestAnimationFrame', function() {
+        it('returns null', function() {
+          expect(SurfNPerf.getFirstPaintFrame()).toEqual(null);
+        });
+      });
+
+      describe('when the client supports window.requestAnimationFrame but it has not yet been called', function() {
+        it('returns undefined', function() {
+          if(!window.requestAnimationFrame) {
+            window.requestAnimationFrame = function() {};
+          }
+          expect(SurfNPerf.getFirstPaintFrame()).toEqual(undefined);
+        });
+      });
+
+      describe('when the client supports window.requestAnimationFrame and it has been called', function() {
```

Comment:
we need a test for each case here:
1. Browser Supports window.requestAnimationFrame and High Resolution Time
2. Browser Supports window.requestAnimationFrame but does not support High Resolution Time